cli/cmdlineparser.cpp

```diff
@@ -321,13 +329,14 @@ bool CmdLineParser::parseFromArgs(int argc, const char* const argv[])
                 mSettings->dump = true;
 
             else if (std::strncmp(argv[i], "--enable=", 9) == 0) {
-                const std::string errmsg = mSettings->addEnabled(argv[i] + 9);
+                const std::string enable_arg = argv[i] + 9;
+                const std::string errmsg = mSettings->addEnabled(enable_arg);
                 if (!errmsg.empty()) {
                     printError(errmsg);
                     return false;
                 }
                 // when "style" is enabled, also enable "warning", "performance" and "portability"
-                if (mSettings->severity.isEnabled(Severity::style)) {
+                if (enable_arg.find("style") != std::string::npos) {
```

Comment:
I don't directly see why we can't check the enum value

Reply:
Because of the left-to-right processing. Note that I am checking the given argument and not the settings.
If you specify `--enable=style` as the hard-coded base and then do `--disable=performance --enable=unusedFunction` it would again re-enable all the implied ones again. I will add a test for that.